File: src/main/java/com/SaasRRHH/main/services/AreaTrabajoService.java
package com.SaasRRHH.main.services;

import com.SaasRRHH.main.model.AreaTrabajo;
import java.util.List;
import java.util.Optional;

public interface AreaTrabajoService {

    List<AreaTrabajo> listar();

    Optional<AreaTrabajo> buscarPorId(Long id);

    AreaTrabajo guardar(AreaTrabajo area);

    void eliminar(Long id);

    Optional<AreaTrabajo> buscarPorNombre(String nombre);

    List<AreaTrabajo> listarActivas();
}

File: frontend/src/app/core/services/area-trabajo.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { AreaTrabajo } from '../models/area-trabajo.model';
import { BaseService } from './base.service';

@Injectable({ providedIn: 'root' })
export class AreaTrabajoService extends BaseService<AreaTrabajo, AreaTrabajo> {
  constructor(http: HttpClient) {
    super(http, 'areas-trabajo');
  }

  listarActivas(): Observable<AreaTrabajo[]> {
    return this.http.get<AreaTrabajo[]>(`${this.baseUrl}/activas`);
  }

  buscarPorNombre(nombre: string): Observable<AreaTrabajo> {
    return this.http.get<AreaTrabajo>(`${this.baseUrl}/buscar`, {
      params: { nombre },
    });
  }
}
